ComponentKit/DataSources/CKComponentBoundsAnimation+UICollectionView.mm

```diff
@@ -33,69 +33,72 @@ @implementation CKComponentBoundsAnimationCollectionViewContext
     // Grab a snapshot view for every cell; they'll be used or discarded in -applyBoundsAnimationToCollectionView:.
-    NSMutableDictionary *sv = [NSMutableDictionary dictionary];
-    for (NSIndexPath *ip in [cv indexPathsForVisibleItems]) {
-      UIView *snapshotView = [[cv cellForItemAtIndexPath:ip] snapshotViewAfterScreenUpdates:NO];
-      if (snapshotView) {
-        sv[ip] = snapshotView;
-      }
-    }
-    _indexPathsToSnapshotViews = sv;
 
-    NSMutableDictionary *la = [NSMutableDictionary dictionary];
-    for (NSIndexPath *ip in visibleAndJustOffscreenIndexPaths(cv)) {
-      la[ip] = [_collectionView layoutAttributesForItemAtIndexPath:ip];
-    }
-    _indexPathsToOriginalLayoutAttributes = la;
-  }
-  return self;
-}
+    const CGSize visibleSize = collectionView.bounds.size;
+    const CGRect visibleRect = { collectionView.contentOffset, visibleSize };
 
-static NSArray *visibleAndJustOffscreenIndexPaths(UICollectionView *cv)
-{
-  NSArray *sortedIndexPaths = [[cv indexPathsForVisibleItems] sortedArrayUsingSelector:@selector(compare:)];
-  if ([sortedIndexPaths count] == 0) {
-    return sortedIndexPaths;
-  }
+    // Obviously we want to animate all visible cells. But what about cells that were not previously visible, but become
+    // visible as a result of an item becoming smaller? We grab the layout attributes of a few more items that are
+    // offscreen so that we can animate them too. (Only some, though; we don't attempt to get *all* layout attributes.)
 
-  // Obviously we want to animate all visible cells. But what about cells that were not previously visible, but become
-  // visible as a result of an item becoming smaller? We grab the layout attributes of a few more items that are
-  // offscreen so that we can animate them too. (Only some, though; we don't attempt to get *all* layout attributes.)
-
-  NSIndexPath *previousIndexPath = [sortedIndexPaths lastObject];
-  static const NSUInteger kOffscreenIndexPathCount = 10;
-  NSMutableArray *offscreenIndexPaths = [NSMutableArray array];
-  while ([offscreenIndexPaths count] < kOffscreenIndexPathCount) {
-    if ([previousIndexPath item] == [cv numberOfItemsInSection:[previousIndexPath section]] - 1) {
-      NSUInteger nextSection = [previousIndexPath section] + 1;
-      while (nextSection < [cv numberOfSections] && [cv numberOfItemsInSection:nextSection] == 0) {
-        nextSection++;
-      }
-      if (nextSection == [cv numberOfSections]) {
-        break; // No more rows to animate.
+    const CGFloat offscreenHeight = visibleSize.height / 2;
+    const CGRect extendedRect = { visibleRect.origin, { visibleSize.width, visibleSize.height + offscreenHeight } };
+
+    NSMutableDictionary *indexPathsToSnapshotViews = [NSMutableDictionary dictionary];
+    NSMutableDictionary *indexPathsToOriginalLayoutAttributes = [NSMutableDictionary dictionary];
```

Comment:
`cellIndexPathsToSnapshotViews` and `cellIndexPathsToOriginalLayoutAttributes` may be a bit clearer.